### PR TITLE
[Dev][Build] .NET 8 Upgrade 

### LINKED ISSUE #1957: Upgrade to .NET 8
### Suggested new feature or improvement

Upgrade to the latest .NET 8 LTS. 

### Scenario

.NET 8 brings much performance enhancements and stability.

### Additional details

https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8

### PR DESCRIPTION
**Pull request by snickler**
## Summary of the pull request
Initial modifications to allow for .NET 8 targeting. 
 
## References and relevant issues
#1957 
## Detailed description of the pull request / Additional comments
- Libraries updated to compatible versions
- Code changes to satisfy Code Analysis warnings
- RuntimeIdentifiers updated to be win-(plat) / RuntimeIdentifiers added to Unit Tests
- Build updated to use .NET 8 SDK

Outstanding issue as of now involves AdaptiveCards, but currently looking into it. 

## PR checklist
- [X] Closes #1957
- [X] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by snickler** [comment]
@crutkas - It says that it's waiting for an approval from a maintainer to kick off.  https://github.com/microsoft/devhome/actions/runs/7021700211

**Comment by crutkas** [comment]
@snickler  is this ready for a review?

**Comment by snickler** [comment]
> @snickler  is this ready for a review?

Not yet. This next commit I'm about to do in the next half hour will make it ready 😎

File: Directory.Build.props
Comment (dhoehna) [inline Directory.Build.props:42]: I remember having to add this when I tried to move to .net8.0.  What is the reason for this again, and can you add a comment explaining it.

File: tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs
Comment (EricJohnson327) [inline tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs:69]: Is this safe?  Do we not need to check for existence before removing?

Comment (snickler) [reply tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs:69]: `Dictionary<TKey,TValue>.Remove returns a boolean. If the key isn't found, it will continue on.

**Review by EricJohnson327 — APPROVED** [review]

**Pull request merged**
